wordpress-mobile/WordPress-Android
# Post Settings: Post Format

This PR updates the Post Format picker and makes several improvements to how it is setup. This is also the first PR that directly updates the `PostModel` after a selection.

We decided to keep the `PostModel` and the DB representation of it up to date in Post Settings. So, in the next PR, I'll update the rest of the settings to dispatch the changes as soon as it happens. Please treat this PR as a work-in-progress still, as there will be couple of changes in the next PR to it. However, aside from how/when the information is saved, it should be close to it's final form.

To test:
* Go into Post Settings and select the Post Format and play with different scenarios. Make sure the correct format is initially selected and when another option is selected, it's updated in the UI correctly.

/cc @tonyr59h
